nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/xml/XMLRecordReader.java

```diff
@@ -487,6 +491,12 @@ private Record parseRecord(StartElement startElement, RecordSchema schema, boole
                     Object value = parseStringForType(content.toString(), contentFieldName, field.get().getDataType());
                     recordValues.put(contentFieldName, value);
                 }
+            } else if (valueElement != null) {

```

Comment:
I would suggest to extract the code or add an in line comment. Later this could be confusing